cns/restserver/restserver.go

```diff
@@ -1426,20 +1429,24 @@ func (service *HTTPRestService) detachNetworkContainerFromNetwork(w http.Respons
 				returnMessage = fmt.Sprintf("[Azure CNS] Error. Network Container %s does not exist.", req.NetworkContainerid)
 				returnCode = NotFound
 			} else {
-				var podInfo cns.KubernetesPodInfo
-				err := json.Unmarshal(existing.CreateNetworkContainerRequest.OrchestratorContext, &podInfo)
-				if err != nil {
-					returnCode = UnexpectedError
-					returnMessage = fmt.Sprintf("Unmarshalling orchestrator context failed with error %+v", err)
-				} else {
-					nc := service.networkContainer
-					pluginBinPath, _ := service.GetOption(acn.OptCNIPath).(string)
-					configPath, _ := service.GetOption(acn.OptCNIConfigFile).(string)
-					netPluginConfig := networkcontainers.NewNetPluginConfiguration(pluginBinPath, configPath)
-					if err = nc.Detach(podInfo.PodName, podInfo.PodNamespace, req.Containerid, netPluginConfig); err != nil {
-						returnMessage = fmt.Sprintf("[Azure CNS] Error. detachNetworkContainerFromNetwork failed %+v", err.Error())
+				switch service.state.OrchestratorType {
+				case cns.Batch:
+					var podInfo cns.KubernetesPodInfo
+					err := json.Unmarshal(existing.CreateNetworkContainerRequest.OrchestratorContext, &podInfo)
```

Comment:
the rest call handlers are becoming too complex from readability point of view. Could you try encapsulating this logic in a new function?